src/utils/settings/index.js

```diff
@@ -123,7 +127,11 @@ function defaults(custom) {
   setupLogger(withDefaults.debug);
 
   // Current ip/hostname information
-  withDefaults.runtime = runtime(withDefaults.core.IPAddressesEnabled,withDefaults.mode === CONSUMER_MODE);
+  withDefaults.runtime = runtime(withDefaults.core.IPAddressesEnabled, withDefaults.mode === CONSUMER_MODE);
+
+  // ensure a valid list of integrations.
+  // `integrations` returns `undefined` or a non-empty array of valid integration items.
+  withDefaults.integrations = integrations(withDefaults);
```

Comment:
Why not always defaulting to have an array, but empty if the data was invalid, so after this point validations become easier?